### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/go-tools/commit/4e8a09d73f5936c6a32c9b25504e0dfb92fedb1b#commitcomment-50570388

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/go-tools/commit/1203a60a51d81065f393707fa453023c7a3c813a

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: analysis/lint/lint.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func parseDirective(s string) (cmd string, args []string) {
 func ParseDirectives(files []*ast.File, fset *token.FileSet) []Directive {
 	var dirs []Directive
 	for _, f := range files {
-		// OPT(dh): in our old code, we skip all the commentmap work if we
+		// OPT(dh): in our old code, we skip all the comment map work if we
```
Comment (jsoref) [inline analysis/lint/lint.go:172]: I know that the function below is `NewCommentMap`, but as this is the only instance of `commentmap`, it doesn't seem worth not having a space...

File: go/ir/doc.go [hunk 2]
```diff
@@ -31,7 +31,7 @@
 //
 // The builder initially builds a naive IR form in which all local
 // variables are addresses of stack locations with explicit loads and
-// stores.  Registerisation of eligible locals and φ-node insertion
+// stores.  Registerization of eligible locals and φ-node insertion
```
Comment (jsoref) [inline go/ir/doc.go:34]: The `z` spelling is used in go/ir/source.go

File: internal/gosmith/context.go [hunk 3]
```diff
@@ -509,7 +509,7 @@ thisPackage:
 }
 
 func (smith *Smith) materializeGotoLabel() string {
-	// TODO: move lavel up
+	// TODO: move label up
```
Comment (jsoref) [inline internal/gosmith/context.go:512]: My initial guess was `level`, but context indicated otherwise

File: internal/gosmith/stmt.go [hunk 4]
```diff
@@ -348,18 +348,18 @@ func (smith *Smith) stmtSwitchExpr() {
 	}
 	// TODO: we generate at most one case, because if we generate more,
 	// we can generate two cases with equal constants.
-	fallth := false
+	fallthru := false
```
Comment (jsoref) [inline internal/gosmith/stmt.go:351]: this spelling is used elsewhere; afaict they're used for the same concept...

File: internal/gosmith/type.go [hunk 5]
```diff
@@ -43,7 +43,7 @@ type Type struct {
 	utyp           *Type   // underlying type
```
Comment (jsoref) [inline internal/gosmith/type.go:43]: `ktyp` says `pointee` type. I decided it was intentional and left it alone. (see below)

File: internal/gosmith/type.go [hunk 6]
```diff
@@ -43,7 +43,7 @@ type Type struct {
...
 	styp           []*Type // function arguments
 	rtyp           []*Type // function return values
-	elems          []*Var  // struct fileds and interface methods
+	elems          []*Var  // struct fields and interface methods
 	literal        func() string
 	complexLiteral func() string
```
Comment (jsoref) [inline internal/gosmith/type.go:49]: the todo below mentions `pointerTo` -- I can't tell if that's related to `pointee` above.

My tooling is much happier with `pointerTo` (even Google Chrome is happy w/ it)

File: lintcmd/cmd.go [hunk 7]
```diff
@@ -217,7 +217,7 @@ func success(allowedChecks map[string]bool, res runner.ResultData) []problem {
 }
 
 func filterIgnored(problems []problem, res runner.ResultData, allowedAnalyzers map[string]bool) ([]problem, error) {
-	couldveMatched := func(ig *lineIgnore) bool {
+	couldHaveMatched := func(ig *lineIgnore) bool {
```
Comment (jsoref) [inline lintcmd/cmd.go:220]: This code is fairly verbose, so I don't see the value in abbreviating this.

I left `could've` as the `'` makes it a word...

I'm happy to drop any commit(s) as requested...

File: unused/testdata/src/linkname/linkname.go [hunk 8]
```diff
@@ -24,4 +24,4 @@ var (
 
 var ol int // used
 
-//go:linkname doesnotexist other5
+//go:linkname nonexistent other5
```
Comment (jsoref) [inline unused/testdata/src/linkname/linkname.go:27]: Afaict the precise term here isn't important...

**Review by jsoref — COMMENTED** [review]
I'm happy to drop any changes, or if changes belong in some upstream (gosmith?), I'm happy to send them along, just let me know.

File: unused/testdata/src/linkname/linkname.go (hunk 8 above)
Comment (dominikh) [reply unused/testdata/src/linkname/linkname.go:27]: > Afaict the precise term here isn't important...

That is true. But it also does not have to be a single, valid English word, either. See the concerns I've expressed on cache_test.go.

**Review by dominikh — CHANGES_REQUESTED** [review]
I wouldn't worry about gosmith's upstream, as that project has been inactive for years.

**Comment by jsoref** [comment]
@dominikh: I've dropped nonexistent. My goal is to improve things in general, I'm not wed to any particular changes.

**Comment by dominikh** [comment]
Merged, thank you!

**Comment by jsoref** [comment]
It sounds like you wouldn't be interested in integrating this tool into your CI. If you change your mind, let me know 😉 .

My goal is to make my tool user-friendly.

**Comment by dominikh** [comment]
> It sounds like you wouldn't be interested in integrating this tool into your CI

I'm afraid not. With something like this, I much prefer the batch approach over something in CI that may unnecessarily impede process on PRs.

**Pull request closed without merging**